### PR TITLE
Reorg to break dependency of `host` package on `enclave` package. Creation of two main methods.

### PR DESCRIPTION
**Pull request by joeldudley**
This does not include the work to create multiple workspaces, which will be part of a follow-up.

The two main methods - host and enclave - can connect to one another. However, they fall over once connected, as the host does not have an L1 node to connect to, and is also not able to communicate with any peers. I thought leaving it this way was preferable for now, rather than replicating the simulation by passing in various fake L1 nodes and peers. We can expand it to make it work later on.

### CODE REVIEW

File: go/obscuronode/nodecommon/types.go [hunk 1]
```diff
@@ -4,17 +4,21 @@ import (
...
 
+	"github.com/ethereum/go-ethereum/core/types"
+
 	"github.com/obscuronet/obscuro-playground/go/hashing"
 	"github.com/obscuronet/obscuro-playground/go/obscurocommon"
 
 	"github.com/ethereum/go-ethereum/common"
 )
 
 // Todo - this has to be a trie root eventually
```
Comment (tudor-malene) [inline go/obscuronode/nodecommon/types.go:15]: this todo should be on the StateRoot type

File: go/obscuronode/nodecommon/enclave.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
+package nodecommon
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/obscuronet/obscuro-playground/go/obscurocommon"
+)
+
+// Enclave - The actual implementation of this interface will call an rpc service
```
Comment (tudor-malene) [inline go/obscuronode/nodecommon/enclave.go:9]: this comment should be change as well, as it does that now

**Review by tudor-malene — CHANGES_REQUESTED** [review]
Thanks Joel.
I had a very quick look. Looks good in principle.
Can you please add a Readme.md in each package explaining what it does?

File: go/obscuronode/nodecommon/types.go (hunk 1 above)
Comment (joeldudley) [reply go/obscuronode/nodecommon/types.go:15]: Addressed.

File: go/obscuronode/nodecommon/enclave.go (hunk 2 above)
Comment (joeldudley) [reply go/obscuronode/nodecommon/enclave.go:9]: Addressed.

**Comment by joeldudley** [comment]
@tudor-malene Addressed your comments and added readmes. The readmes are based on my best understanding and it would be good to validate them. I think there are various classes and functions that aren't in exactly the right place given the readme descriptions - we should probably fix this in the future.

File: go/obscuronode/enclave/README.md [hunk 3]
```diff
@@ -0,0 +1 @@
+This package contains code related to the node's enclave component.
```
Comment (tudor-malene) [inline go/obscuronode/enclave/README.md:1]: Umm. I think we should elaborate more here. 
Talk a bit how this has to be compiled with ego, started in a separate process, from the main method, etc

File: go/obscuronode/nodecommon/README.md [hunk 4]
```diff
@@ -0,0 +1 @@
+This package contains code that is shared between the node host and the node enclave.
```
Comment (tudor-malene) [inline go/obscuronode/nodecommon/README.md:1]: also maybe a bit more elaboreated.
This is basically the API to the enclave

**Review by tudor-malene — CHANGES_REQUESTED** [review]

File: go/obscuronode/enclave/main/main.go [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+	nodeAddressBytes, port := parseCLIArgs()
+
+	nodeAddress := common.BytesToAddress([]byte(*nodeAddressBytes))
+	if _, err := enclave.StartServer(*port, nodeAddress, nil); err != nil {
+		panic(err)
+	}
+
+	fmt.Printf("Enclave server listening on port %d.\n", *port)
+	select {}
```
Comment (otherview) [inline go/obscuronode/enclave/main/main.go:47]: What if there was a graceful stop mechanism? Would a chan that this select listens on, work ?

**Review by otherview — CHANGES_REQUESTED** [review]
The refactor of the packages looks good. However I wouldn't include the obscuronode cmd and enclave cmd if they aren't working - Probably worth including those with the P2P PR instead.

File: go/obscuronode/enclave/main/main.go (hunk 5 above)
Comment (joeldudley) [reply go/obscuronode/enclave/main/main.go:47]: But what would send to that chan? This is being interrupted by some external action (ctrl + c).

**Pull request merged**